rewrite-core/src/main/java/org/openrewrite/text/FindAndReplace.java

```diff
@@ -99,6 +102,39 @@ public String getDescription() {
     @Nullable
     String filePattern;
 
+    @Option(displayName = "Plaintext only", description = "Only alter files that are parsed as plaintext to prevent language-specific LST information loss. Defaults to false.",
+            required = false)
+    @Nullable
+    Boolean plaintextOnly;
+
+    @Deprecated
+    public FindAndReplace(final String find, @Nullable final String replace, @Nullable final Boolean regex,
+            @Nullable final Boolean caseSensitive, @Nullable final Boolean multiline, @Nullable final Boolean dotAll,
+            @Nullable final String filePattern) {
+        this.find = find;
+        this.replace = replace;
+        this.regex = regex;
+        this.caseSensitive = caseSensitive;
+        this.multiline = multiline;
+        this.dotAll = dotAll;
+        this.filePattern = filePattern;
+        this.plaintextOnly = null;
+    }
+
+    @JsonCreator
```

Comment:
```suggestion
```

Reply:
While I appreciate your concerns for backwards compatibility, I think it's fine in this case to add the argument and just fix it downstream if needed. I reality folks will mostly use this recipe from Yaml files anyway, such that there's rarely a need to update code.